Bottom: 15f1fec5ad24dcb632495349f5d850215fe55577
Top:    30fdb019b5be0d82c7810ccedccb6db7405bef90
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-26 22:41:45 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 679c32d..2e8444e 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -230,6 +230,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
